src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -198,9 +206,23 @@ DecodedResults StaticLLMPipeline::generate(
     }
 
     OPENVINO_ASSERT(std::holds_alternative<std::string>(inputs));
-    auto tokenized_input = m_tokenizer.encode(std::get<std::string>(inputs));
+    auto& prompt = std::get<std::string>(inputs);
+
+    if (m_is_chat_conversation) {
+        m_history.push_back({{"role", "user"}, {"content", prompt}});
+        constexpr bool add_generation_prompt = true;
+        prompt = m_tokenizer.apply_chat_template(m_history, add_generation_prompt);
+    }
+
+    auto tokenized_input = m_tokenizer.encode(prompt);
     auto encoded_results = generate(tokenized_input, config, streamer);
-    return {m_tokenizer.decode(encoded_results.tokens), encoded_results.scores};
+    DecodedResults decoded_results = {m_tokenizer.decode(encoded_results.tokens), encoded_results.scores};
```

Comment:
Just to make it clear. Does that implementation runs the whole history on every generate()?

Reply:
Yes, 
```
EncodedResults StaticLLMPipeline::generate(
    const EncodedInputs& inputs,
    OptionalGenerationConfig generation_config,
    StreamerVariant streamer)
```
knows nothing about chat conversation, it consumes input prompt up to 1024 tokens

Reply:
Why only 1024? Shouldn't this be defined by the model? Also, when we turn the model to static shape form, what we use as a guidance to fix the input prompt/kvcache size?

Reply:
> Shouldn't this be defined by the model?
It shouldn't model has dynamic shape